bip-tx-terminology.mediawiki

```diff
@@ -0,0 +1,286 @@
+</pre>
+
+== Introduction==
+
+===Abstract===
+
+This document describes terminology for referring to scripts, transaction parts, and serialization artifacts in the context of transactions.
+
+===Copyright===
+
+This document is licensed under the 3-clause BSD license.
+
+===Motivation===
+
+In the past years, we have seen many discussions about leveraging Taproot to facilitate new spending conditions.
+We noticed that terminology to refer to the components of transactions and functions that scripts fulfill in transactions includes a broad spectrum of alternative jargon.
+In this document, we describe a set of terms that can be used to refer to transaction components by their function or their context.
+The terms are not entirely consistent in themselves since there are many terms that are already firmly established.
+Instead of striving for consistency, we try to relate terms to each other, explore the origins of terms, and fill in blanks.
+
+===Objects and their Serialization===
+
+We treat the objects of the Bitcoin Protocol (e.g. Transactions, Blocks) as distinct from their serialization.
+We distinguish concrete properties of transactions such as fields and scripts (labeled [TC] for Transaction Component) from artifacts of the transaction serialization (labeled [SA] for Serialization Artifact).
+We also introduce some concepts and ideas that are useful to describe aspects of transcations (labeled [AT] for Abstract Term).
+For example, version and locktime are fields that can be set during transaction creation and therefore properties of a transaction, while the input counter is a serialization artifact resulting from how the input list of the transaction is represented.
+Likewise, the _input script_ is a transaction field, but the length indicator of the input script a serialization artifact.
+
+===Definition of Terms===
+
+; Annex
+: [TC] A Witness Item reserved for future extensions, currently unused (see [[https://github.com/bitcoin/bips/blob/master/bip-0341.mediawiki#cite_note-5|BIP341]]).
+
+; Amount
+: [TC] The count of satoshis assigned to an output.
+
+; Bare Output Script
+: [AT] A Condition Script that is directly defined in the Output Script field. Used to specify all types of Bare Outputs including Pay to Pubkey, Pay to Public Key Hash, Pay to Multisig outputs, and arbitrary scripts.
+
+; Condition Script
+: [AT] Collective term for scripts that are not forwarding scripts and encode a set of conditions that could be fulfilled to spend a UTXO.  A scriptPubkey can either contain directly a condition script in the case of a bare output, or will commit to one or multiple condition scripts for other output types.  For P2TR the tweaked public key is considered a condition script as well as each Leaf Script in the script tree.  At the time of writing, there are four distinct Condition Scripts: Witness Script, Leaf Script, Redeem Script (only non-segwit P2SH), and Bare Output Script (bare outputs, Pay to Multisig, P2PK, P2PKH).
```

Comment:
Thanks, I’ve adopted your phrasing of one or multiple conditions. Even before Taproot, a script can have multiple ways to execute, e.g. in an HTLC output it can be resolved either via the timelock or via the pre-image.